src/crypto/common.h

```diff
@@ -5,62 +5,98 @@
 #ifndef BITCOIN_CRYPTO_COMMON_H
 #define BITCOIN_CRYPTO_COMMON_H
 
-#if defined(HAVE_CONFIG_H)
-#include "bitcoin-config.h"
-#endif
-
 #include <stdint.h>
 
-#include "compat/endian.h"
-
 uint16_t static inline ReadLE16(const unsigned char* ptr)
 {
-    return le16toh(*((uint16_t*)ptr));
+    return ((uint16_t)(ptr[0])) |
+           (((uint16_t)(ptr[1])) << 8);
 }
 
 uint32_t static inline ReadLE32(const unsigned char* ptr)
 {
-    return le32toh(*((uint32_t*)ptr));
+    return ((uint32_t)(ptr[0])) |
+           (((uint32_t)(ptr[1])) << 8) |
+           (((uint32_t)(ptr[2])) << 16) |
+           (((uint32_t)(ptr[3])) << 24);
 }
 
 uint64_t static inline ReadLE64(const unsigned char* ptr)
 {
-    return le64toh(*((uint64_t*)ptr));
+    return ((uint64_t)(ptr[0])) |
```

Comment:
Seems ARM supports unaligned reads pretty much since the architecture we support.